pkg/streamingpromql/operators/name_drop.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package operators
+
+import (
+	"context"
+
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+	"github.com/grafana/mimir/pkg/util/limiter"
+)
+
+type NameDrop struct {
```

Comment:
[nit] `DropName` sounds more natural to me than `NameDrop` (similar to how `DeduplicateAndMerge` starts with a verb`